### DISCUSSION TITLE
Using Page properties in templates

### DISCUSSION THREAD

**Discussion by pomealan** [microsoft-search/pnp-modern-search#1567] [Q&A] [answered]
The documentation notes that you can use page properties in the template using handlebar helpers but this does not appear to work in my template.
The format is as follows:
{{getPageContext "<SPFx page property>"}} the documentation only shows example for user and language but assume page properties are supported. {{getPageContext "user.displayName"}} or {{getPageContext "cultureInfo.currentUICultureName"}}
I'm using this to create a URL to perform do a drill down type search based on the page property:
{{getPageContext "Page.Product.Label"}}

https://microsoft-search.github.io/pnp-modern-search/extensibility/templating/

**Comment by edarroudi** [comment, marked as answer]
Hi @pomealan , if you switch to debug layout  in the results webpart you'll see in the "context" property all properties that you can use. I didn't find any Page.Product.Label...

**Comment by pomealan** [comment]
I found the context and can see the properties are grouped under site, web, user, list, cultureinfo and listitem. There is no Page context in that listing which explains why I can't get the property for the page meta data.
I read the documentation as providing access to page properties as in "page property" that I read a page meta data:
{{getPageContext "<SPFx page property>"}}

Thanks for clearing this up. As page meta data can be retrieved as tokens in the query template could this be implemented for use in the templates?

**Reply by edarroudi** [reply]
I think there are several possibilities for that, at least for something like the Page Properties.
You have to convince @FranckyC  or @wobba  that there is a need and find someone to implement it (or do it yourself).
You can write an custom Handlebar Helper for it as well, probably only copying the code from the tokenservice

**Comment by wobba** [comment]
@pomealan does using `getPageContext` work for you? https://docs.microsoft.com/en-us/javascript/api/sp-page-context/pagecontext?view=sp-typescript-latest lists all the objects available. You have `listItem` available as an example, which is backing any page.
